var/spack/repos/builtin/packages/hdf-eos2/package.py

```diff
@@ -160,4 +160,15 @@ def configure_args(self):
         if "zlib" in self.spec:
             extra_args.append("--with-zlib={0}".format(self.spec["zlib-api"].prefix))
 
+        # https://forum.hdfgroup.org/t/help-building-hdf4-with-clang-error-implicit-declaration-of-function-test-mgr-szip-is-invalid-in-c99/7680
+        # -fPIC: https://github.com/spack/spack/issues/43792
+        if self.spec.satisfies("%apple-clang"):
+            extra_args.append(
+                "CFLAGS=-Wno-error=implicit-function-declaration {0}".format(
+                    self.compiler.cc_pic_flag
+                )
+            )
+        else:
+            extra_args.append("CFLAGS={0}".format(self.compiler.cc_pic_flag))
```

Comment:
For future reference, we are moving the core to use Python f-strings (https://docs.python.org/3/tutorial/inputoutput.html) and encouraging the same in packages.